### DISCUSSION TITLE
Is there a way to not invalidate the cache if only the `version` field of the package.json file changes?

### DISCUSSION THREAD

**Discussion by gablabelle** [nrwl/nx#12421] [General]
Hello,

We have a monorepo using `pnpm`, `lerna` and `nx cloud` with the following structure:

```
apps/**
modules/** (private packages, not versioned nor published)
libs/** (private packages, not versioned nor published)
```

Everytime we merge into our development branch we bump the versions our of `apps/*/package.json` and deployed those to our dev environment.

The problem is that this causes the cache to miss on these apps since the `apps/*/package.json` files has changed. So we added the following to the `.nxignore` file:

```
apps/*/package.json
pnpm-lock.yaml
```

But then if someone adds/updates a dependency in `apps/*/package.json` the cache doesn't get invalidate as we would like.

So the question is: Is there a way to **not** invalidate the cache if only the `version` field of the `apps/*/package.json` files changes?

**Comment by twittwer** [comment]
Maybe you can configure your `implicitDependencies`. There you can define which parts of the package.json are relevant.
https://nx.dev/reference/nx-json#files-&-implicit-dependencies

**Reply by gablabelle** [reply]
> Maybe you can configure your `implicitDependencies`. There you can define which parts of the package.json are relevant.
> 
> https://nx.dev/reference/nx-json#files-&-implicit-dependencies

Thanks a lot! It looks like it's exactly what i was looking for. 👍🏽
